### ISSUE TITLE
unable to re-schedule job at specific time

### ISSUE THREAD

**Issue by lksv** [mhenrixon/sidekiq-unique-jobs#108]
In worker, I would like to postpone processing of job for some period of time.

e.g.:

```ruby
class DocumentPlainTextWorker
  include Sidekiq::Worker
  sidekiq_options unique: true, queue: 'plain_text_getter'

  EXTERNAL_TIME_TO_OCR = 2.hours

  def perform(document_id)
    document = Document.lock.find(document_id)
    return if document.already_processed?

    document.text_storage = Dragonfly.app.fetch_url(document.external_ocr_result_url)
    max_time = document.created_at + EXTERNAL_TIME_TO_OCR
    if !document.already_processed? && (max_time > Time.now)
      self.class.perform_at(max_time)
      return
    end
   
    ...make my own ocr here...
end
```
  
The problem is that the option "uniq" deny of creating same task later.

...maybe this is a new feature or just wandering how to do it with `uniq:true` option.

**Comment by mhenrixon** [comment]
I'm getting to that in a few days! If you use the master branch there is a run lock that should allow you to schedule any number of jobs but only lock them during run.

**Comment by mhenrixon** [comment]
Should be fixed in 4.0.0; report back if not!
